app/controllers/reading_groups_controller.rb

```diff
@@ -56,19 +56,17 @@ def destroy
 
   private
 
-    def set_reading_group
-      begin
-        @reading_group = ReadingGroup.find(params[:id].to_i)
-        true
-      rescue ActiveRecord::RecordNotFound => exception
-        respond_to do |format|
-          format.json { render json: {errors: [exception.message]}, status: :not_found }
-        end
-        false
-      end
+  def set_reading_group
+    @reading_group = ReadingGroup.find(params[:id].to_i)
+    true
+  rescue ActiveRecord::RecordNotFound => exception
```

Comment:
I'm seeing a lot of this `rescue` block. What do you think if we turned it into a `rescue_from` clause on `ApplicationController`?